components/list/list-item-mixin.js

```diff
@@ -0,0 +1,351 @@
+import ResizeObserver from 'resize-observer-polyfill';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+
+const ro = new ResizeObserver(entries => {
+	entries.forEach(entry => {
+		if (!entry || !entry.target || !entry.target.resizedCallback) {
+			return;
+		}
+		entry.target.resizedCallback(entry.contentRect && entry.contentRect.width);
+	});
+});
+
+const defaultBreakpoints = [842, 636, 580, 0];
+
+export const ListItemMixin = superclass => class extends ListItemDragDropMixin(ListItemCheckboxMixin(ListItemRoleMixin(RtlMixin(superclass)))) {
+
+	static get properties() {
+		return {
+			/**
+			 * Breakpoints for responsiveness in pixels. There are four different breakpoints and only the four largest breakpoints will be used.
+			 */
+			breakpoints: { type: Array },
+			/**
+			 * Address of item link if navigable
+			 */
+			href: { type: String },
+			_breakpoint: { type: Number },
+			_dropdownOpen: { type: Boolean, attribute: '_dropdown-open', reflect: true },
+			_hoveringLink: { type: Boolean },
+			_focusing: { type: Boolean },
+			_focusingLink: { type: Boolean },
+			_tooltipShowing: { type: Boolean, attribute: '_tooltip-showing', reflect: true }
+		};
+	}
+
+	static get styles() {
+
+		const styles = [ css`
+			:host {
+				display: block;
+				position: relative;
+				margin-top: -1px;
+			}
+			:host[hidden] {
+				display: none;
+			}
+			:host([href]) {
+				--d2l-list-item-content-text-color: var(--d2l-color-celestine);
+			}
+			:host([_tooltip-showing]),
+			:host([_dropdown-open]) {
+				z-index: 10;
+			}
+			:host(:first-child) d2l-list-item-generic-layout[data-separators="between"] {
+				border-top: 1px solid transparent;
+			}
+			:host(:last-child) d2l-list-item-generic-layout[data-separators="between"] {
+				border-bottom: 1px solid transparent;
+			}
+			:host([dragging]) d2l-list-item-generic-layout {
+				filter: grayscale(75%);
+				opacity: 0.4;
+			}
+			.d2l-list-item-drag-image {
+				transform: rotate(-1deg);
+			}
+			d2l-list-item-generic-layout {
+				background: white;
+				transform: rotate(1deg);
+			}
+			d2l-list-item-generic-layout[data-separators="none"] {
+				border-top: 1px solid transparent;
+				border-bottom: 1px solid transparent;
+			}
+			d2l-list-item-generic-layout {
+				border-bottom: 1px solid transparent;
+				border-bottom: 1px solid var(--d2l-color-mica);
+				border-top: 1px solid var(--d2l-color-mica);
+			}
+			[slot="content"].d2l-list-item-content-extend-separators {
+				padding-left: 0.9rem;
+				padding-right: 0.9rem;
+			}
+			a[href].d2l-list-item-link {
+				width: 100%;
+				height: 100%;
+			}
+			.d2l-list-item-content ::slotted(*) {
+				margin-top: 0.05rem;
+			}
+			.d2l-list-item-content.d2l-hovering,
+			.d2l-list-item-content.d2l-focusing {
+				--d2l-list-item-content-text-decoration: underline;
+			}
+			[slot="content-action"]:focus {
+				outline: none;
+			}
+			[slot="content"] {
+				justify-content: stretch;
+				padding: 0.55rem 0px;
```

Comment:
Nit: no units on `0` values.  Did the style linter not catch this?

Reply:
Oh, the list components are still being ignored for style linting.  We should get that enabled as soon as convenient.

Reply:
I'll enable that and every time. You think I would catch this by now lol.